samples/ingestion/ingestion-client/FetchTranscription/Language/AnalyzeConversationsProvider.cs

```diff
@@ -0,0 +1,255 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
+// </copyright>
+
+namespace FetchTranscription.Language
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text.Json;
+    using System.Threading.Tasks;
+    using Azure;
+    using Azure.AI.Language.Conversations;
+    using Azure.Core;
+    using Connector;
+    using Connector.Serializable.Language.Conversations;
+    using FetchTranscriptionFunction;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    /// <summary>
+    /// Analyze Conversations async client.
+    /// </summary>
+    public class AnalyzeConversationsProvider
+    {
+        private static readonly TimeSpan RequestTimeout = TimeSpan.FromMinutes(3);
+
+        private readonly ConversationAnalysisClient ConversationAnalysisClient;
+
+        private readonly string Locale;
+
+        private readonly ILogger Log;
+
+        public AnalyzeConversationsProvider(string locale, string subscriptionKey, string region, ILogger log)
+        {
+            ConversationAnalysisClient = new ConversationAnalysisClient(new Uri($"https://{region}.api.cognitive.microsoft.com"), new AzureKeyCredential(subscriptionKey));
+
+            Locale = locale;
+            Log = log;
+        }
+
+        public bool IsConversationalPiiEnabled()
+        {
+            return Locale.Contains("en", StringComparison.OrdinalIgnoreCase) && FetchTranscriptionEnvironmentVariables.ConversationPiiSetting != Connector.Enums.ConversationPiiSetting.None;
+        }
+
+        /// <summary>
+        /// API to submit an analyzeConversations async Request.
+        /// </summary>
+        /// <param name="speechTranscript">Instance of the speech transcript.</param>
+        /// <returns>An enumerable of the jobs IDs and errors if any.</returns>
+        public async Task<(IEnumerable<string> jobIds, IEnumerable<string> errors)> SubmitAnalyzeConversationsRequestAsync(SpeechTranscript speechTranscript)
+        {
+            speechTranscript = speechTranscript ?? throw new ArgumentNullException(nameof(speechTranscript));
+
+            var data = new AnalyzeConversationsRequest
+            {
+                DisplayName = "IngestionClient",
+                AnalysisInput = new AnalysisInput(new[]
+                {
+                    new Conversation
+                    {
+                        Id = Guid.NewGuid().ToString(),
+                        Language = Locale,
+                        Modality = Modality.transcript,
+                        ConversationItems = speechTranscript.RecognizedPhrases.Where(r => !string.IsNullOrEmpty(r.NBest.First().Lexical))
+                            .Select(item => new ConversationItem()
+                            {
+                                Text = item.NBest.First().Display,
```

Comment:
nit: you could get item.NBest.First() once in the select, then you don't need to call First() so many times

Reply:
fixed